### PR TITLE
Added email verification auto-redirect

### PR DESCRIPTION
**Pull request by micwu562**
Worked on signup email verification:
- clicking on the email verification link and returning to the canopy tab should auto redirect you to login
- changed some text on the page and refactored code (hopefully it's ok)

### CODE REVIEW

File: web/pages/verify.tsx [hunk 1]
```diff
@@ -90,18 +107,22 @@ function VerifyYourEmail() {
...
+                if (currentUser) {
+                  setLoadingResendVerification(true);
+                  await sendEmailVerification(currentUser)
+                    .then(() => {
+                      toast.success("Verification email re-sent!");
+                    })
+                    .catch((error) => {
+                      toast.error(
+                        error.message.includes("too-many-requests")
+                          ? "Please wait until sending another verification email."
```
Comment (legitmaxwu) [inline web/pages/verify.tsx:119]: wait how long?

File: web/pages/verify.tsx [hunk 2]
```diff
@@ -21,38 +22,52 @@ function VerifyYourEmail() {
...
+
+      await fetch(`/api/auth/upsertUserData`, {
+        method: "POST",
+        headers: {
+          authorization: `Bearer ${await currentUser.getIdToken()}`,
+        },
+      })
+        .then(async () => {
+          // wait half a second before redirecting to show verified screen
+          await new Promise((resolve) => setTimeout(resolve, 500));
```
Comment (legitmaxwu) [inline web/pages/verify.tsx:51]: perhaps create a util function called sleep and then `await sleep(500)`

**Review by legitmaxwu — APPROVED** [review]
looks good overall!

File: web/pages/verify.tsx (hunk 1 above)
Comment (micwu562) [reply web/pages/verify.tsx:119]: Not sure how long it is

**Pull request merged**